### ISSUE TITLE
Provide support for metrics through OpenTelemetry 

### ISSUE THREAD

**Issue by tomkerkhove** [kedacore/keda#2291]
### Proposal

Provide support for metrics through [OpenTelemetry](https://opentelemetry.io/docs/concepts/what-is-opentelemetry/) which allows end-users to use the destination that they prefer.

![image](https://user-images.githubusercontent.com/4345663/141951216-dcb998cd-d1c4-4f4c-96cc-35e420d9e514.png)

_Source: https://opentelemetry.io/docs/_

Ideally, end-users can use the OTLP collector and take it from there.

### Use-Case

Allows end-users to use the destination that they prefer instead of us having to implement every one of them.

We already support Prometheus, and HTTP add-on is adding support through https://github.com/kedacore/http-add-on/issues/321, but this allows us to be more flexible.

### Anything else?

Metrics in OpenTelemetry spec has just gone stable (https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/api.md) but some pieces are still experimental so we should wait (https://github.com/open-telemetry/opentelemetry-specification/tree/main/specification/metrics)

Go support is in Alpha (https://opentelemetry.io/docs/go/#status-and-releases) but already has decent coverage (https://github.com/open-telemetry/opentelemetry-specification/blob/main/spec-compliance-matrix.md#metrics)

Long story short, already tracking for when it goes stable.

**Comment by tomkerkhove** [comment]
Duplicate of https://github.com/kedacore/keda/issues/3078
